### PR TITLE
* NEW [mqtt_parser_test] Add tests for mqtt_parser.c.

### PR DESCRIPTION
**Pull request by Hermann0222**

### CODE REVIEW

**Comment by JaylinYu** [comment]
nice try

File: src/sp/protocol/mqtt/mqtt_parser_test.c [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+#include <stdio.h>
+
+static int
+test_utf8_check()
+{
+	int     rv;
+	uint8_t src[] = { 0x24, 0x4D, 0x51, 0x54, 0x54 };
+	// TODO more cases for failure.
+	rv = utf8_check((char *) src, strlen((char *) src) - 1);
+	assert(rv == ERR_SUCCESS);
```
Comment (JaylinYu) [inline src/sp/protocol/mqtt/mqtt_parser_test.c:12]: try a opposite test also? input malicious data

**Review by JaylinYu — APPROVED** [review]

**Comment by JaylinYu** [comment]
CI failed

**Review by JaylinYu — CHANGES_REQUESTED** [review]
CI failed

**Comment by Hermann0222** [comment]
It shoud work now.

File: src/sp/protocol/mqtt/mqtt_parser_test.c (hunk 1 above)
Comment (Hermann0222) [reply src/sp/protocol/mqtt/mqtt_parser_test.c]: ok.

**Review by JaylinYu — APPROVED** [review]
well done

**Pull request merged**